### PR TITLE
Deprecation of 2.11, parallel and addition of optics syntax

### PR DESCRIPTION
**Pull request by Odomontois**

### CODE REVIEW

**Comment by Z1kkurat** [comment]
Since this PR has some major changes (bumping cats version, removing deprecated typeclasses), it would be nice to rename it so the name could reflect PR's content (or at least add some description to it).

File: env/src/main/scala/tofu/env/Env.scala [hunk 1]
```diff
@@ -184,7 +184,9 @@ sealed trait Env[E, +A] {
   def executeOn(s: Scheduler, forceAsync: Boolean = true): Env[E, A] =
     mapTask(_.executeOn(s, forceAsync))
   def fork: Env[E, Fiber[Env[E, *], A @uv]] = executeAsync.start
-  def forkAndForget: Env[E, Unit]           = mapTask(_.forkAndForget)
+  @scala.deprecated("Replaced with startAndForget", "0.2.1")
```
Comment (Z1kkurat) [inline env/src/main/scala/tofu/env/Env.scala]: Shouldn't it be '0.3.0'?

File: env/src/main/scala/tofu/env/EnvTraversing.scala [hunk 2]
```diff
@@ -1,66 +1,64 @@
 package tofu.env
 
 import monix.eval.Task
-
-import scala.collection.TraversableLike
-import scala.collection.generic.CanBuildFrom
+import monix.execution.compat.BuildFrom
```
Comment (danslapman) [inline env/src/main/scala/tofu/env/EnvTraversing.scala]: I think we should use official `scala.collection.compat.BuildFrom` instead

Comment (Odomontois) [reply env/src/main/scala/tofu/env/EnvTraversing.scala]: I can't see how we might use it having monix API is built upon that

Comment (Odomontois) [reply env/src/main/scala/tofu/env/EnvTraversing.scala]: I'll try

Comment (Odomontois) [reply env/src/main/scala/tofu/env/EnvTraversing.scala]: No. Scala 2.12.9 wont build

Comment (danslapman) [reply env/src/main/scala/tofu/env/EnvTraversing.scala]: Okay, sad though

**Pull request merged**